kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsSystem.scala

```diff
@@ -33,29 +31,17 @@ import org.apache.kyuubi.service.CompositeService
 class MetricsSystem extends CompositeService("MetricsSystem") {
 
   private val registry = new MetricRegistry
-  private var counters: LoadingCache[String, Counter] = _
-  private val countersLock = new ReentrantLock
-
-  def incAndGetCount(key: String): Long = {
-    try {
-      countersLock.lock()
-      val counter = counters.get(key)
-      counter.inc(1L)
-      counter.getCount
-    } finally {
-      countersLock.unlock()
-    }
+
+  def incAndGetCount(key: String): Long = synchronized {
+    val counter = registry.counter(key)
+    counter.inc(1L)
+    counter.getCount
```

Comment:
The return value never used, if return `Unit`, we can remove `synchronized`

Reply:
sgtm